src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData)
+      result.push({ ...newState });
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach(key => {
+        delete newState[key];
+      });
+
+      result.push(Object.assign({}, newState));
+    }
```

Comment:
```suggestion
result.push({ ...newState });    
```

Reply:
It's the same.
You can move this .push from every iteration and add just one, at the end of loop